// gulp-message/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const util = require("gulp-util");
const chalk = require("chalk");
/**
 * Transform a chunk that may be a standalone stream into a MessageChunk.
 */
const createChunk = (x) => typeof x === 'string' ? [x, chalk.reset] : x;
/**
 * Render a collection of message chunks into a single, printable string.
 */
const render = (...c) => c.map(([msg, style]) => style(msg))
    .filter((x) => x.length)
    .join(' ');
/**
 * Create a styled, prefixed logger.
 */
exports.logger = (opts) => (message) => {
    const chunks = [
        opts.prefix || '',
        [message, opts.style || chalk.reset]
    ].map(createChunk);
    (opts.writer || util.log)(render(...chunks));
};
/**
 * Emit an error message.
 */
exports.error = exports.logger({
    prefix: ['ERROR', chalk.white.bgRed]
});
/**
 * Emit a warning message.
 */
exports.warn = exports.logger({
    prefix: ['WARNING', chalk.black.bgYellow]
});
/**
 * Emit an info message.
 */
exports.info = exports.logger({});
/**
 * Emit a debug message.
 */
exports.debug = exports.logger({
    prefix: '→',
    style: chalk.dim
});
